detectron2/data/datasets/coco.py

```diff
@@ -280,6 +286,131 @@ def file2id(folder_path, file_path):
+                segmentation = annotation["segmentation"]
+                # TODO: check segmentation type: RLE, BinaryMask or Polygon
+                polygons = PolygonMasks([segmentation])
+                area = polygons.area()[0].item()
+            else:
+                # Computing areas using bounding boxes
+                area = Boxes([bbox]).area()[0].item()
+
+            # COCO requirement:
+            #   linking annotations to images
+            #   "id" field must start with 1
+            coco_annotation["id"] = len(coco_annotations) + 1
+            coco_annotation["image_id"] = image_dict["image_id"]
+            coco_annotation["bbox"] = bbox
+            coco_annotation["area"] = area
+            coco_annotation["category_id"] = annotation["category_id"]
+            coco_annotation["iscrowd"] = annotation.get("iscrowd", 0)
+
+            # Add optional fields
+            if "keypoints" in annotation:
+                coco_annotation["keypoints"] = annotation["keypoints"]
+            if "num_keypoints" in annotation:
+                coco_annotation["num_keypoints"] = annotation["num_keypoints"]
+            if "segmentation" in annotation:
+                coco_annotation["segmentation"] = annotation["segmentation"]
+
+            coco_annotations.append(coco_annotation)
+
+    logger.info(
+        "Conversion finished, "
+        f"num images: {len(coco_images)}, num annotations: {len(coco_annotations)}"
+    )
+
+    info = {
+        "date_created": str(datetime.datetime.now()),
+        "description": "Automatically generated COCO json file for Detectron2.",
+    }
+    coco_dict = {
+        "info": info,
+        "images": coco_images,
+        "annotations": coco_annotations,
+        "categories": categories,
+        "licenses": None,
+    }
+    return coco_dict
+
+
+def convert_to_coco_json(dataset_name, output_folder="", allow_cached=True):
+    """
+    Converts dataset into COCO format and saves it to a json file.
+
+    Args:
+        dataset_name: reference from the config file to the catalogs
+        output_folder: where json file will be saved and loaded from
+        allow_cached: if json file is already present then skip conversion
+    Returns:
+        cache_path: path to the COCO-format json file
+    """
+
+    # TODO: The dataset or the conversion script *may* change,
+    # a checksum would be useful for validating the cached data
+    cache_path = os.path.join(output_folder, f"{dataset_name}_coco_format.json")
+    os.makedirs(output_folder, exist_ok=True)
+    if os.path.exists(cache_path) and allow_cached:
+        logger.warning(f"Reading cached annotations in COCO format from:{cache_path}")
+    else:
+        logger.info("Fetching dataset annotations")
+        coco_dict = convert_to_coco_dict(dataset_name)
+
+        # TODO: json_file should be renamed to json_path
+        with open(cache_path, "w") as json_file_buff:
```

Comment:
use `PathManager.open` so that it works with different file storage backends that are not necessarily a local file system.

Also I don't quite get the TODO here. If it's about the name "json_file_buff", I'm ok with just "f" or "buf" since it's clear enough from the context

Reply:
Modified to use PathManager.

moved TODO to coco_evaluation (the comment was misplaced). `json_file` should be  `json_path`.